### PR TITLE
Add expert parallelism support in JaxFusedMoE.

### PR DESCRIPTION
**Pull request by hfan**
# Description

Shard JaxFusedMoE weight tensors by the expert dim and use `ragged_all_to_all` to send/recv the `gmm` results from all experts.

# Tests

Ran the Qwen3-30B-A3B and Mixtral-8x7B models, note the `--enable-expert-parallel` command line argument:

```
MODEL_IMPL_TYPE=vllm TPU_BACKEND_TYPE=jax python examples/offline_inference/basic/generate.py     --model=Qwen/Qwen3-30B-A3B     --tensor_parallel_size=4     --task=generate     --max_model_len=1024     --max_num_seqs=1 --enable-expert-parallel
```

```
MODEL_IMPL_TYPE=vllm TPU_BACKEND_TYPE=jax python examples/offline_inference/basic/generate.py     --model=mistralai/Mixtral-8x7B-Instruct-v0.1     --tensor_parallel_size=8     --task=generate     --max_model_len=1024     --max_num_seqs=1  --enable-expert-parallel
```

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ ] I have performed a self-review of my code.
- [ ] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ ] I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_commons/models/vllm/jax_fused_moe.py [hunk 1]
```diff
@@ -218,6 +381,7 @@ def __init__(self, fused_moe: torch.nn.Module, mesh: Mesh):
         self.global_num_experts = fused_moe.global_num_experts
         self.renormalize = fused_moe.renormalize
         self.reduce_results = fused_moe.reduce_results
+        self.use_ep = vllm_parallel_config.enable_expert_parallel
```
Comment (lsy323) [inline tpu_commons/models/vllm/jax_fused_moe.py:384]: We can get `use_ep` from the `FusedMoE` layer directly https://github.com/vllm-project/vllm/blob/8b6e1d639c66d5828d03a7df2c3a500030a5c5cd/vllm/model_executor/layers/fused_moe/layer.py#L865. So that we don't need to pipe the vllm parallel config through.

**Comment by lsy323** [comment]
Left a comment to avoid piping parallel config through the module swap workflow. Otherwise lgtm!

File: tpu_commons/models/vllm/jax_fused_moe.py (hunk 1 above)
Comment (hfan) [reply tpu_commons/models/vllm/jax_fused_moe.py:384]: I thought about that too, but then the use_ep in FusedMoEParallelConfig is determined also by tp_size and dp_size, which we overwrite as 1 in order to load the vllm model into CPU before moving and sharding the model to TPU: https://github.com/vllm-project/vllm/blob/8b6e1d639c66d5828d03a7df2c3a500030a5c5cd/vllm/model_executor/layers/fused_moe/layer.py#L171

So I figured more reliable way to get the flag is from the config
